tests/test_datasets.py

```diff
@@ -47,6 +47,101 @@ def cwd(path):
+
+
+class TestFileSystemDataset(unittest.TestCase):
+    @contextmanager
+    def _setup_storage(self) -> Iterator[Tuple[FileSystemDataStorage, WickerPathFactory, str]]:
+        with tempfile.TemporaryDirectory() as tmpdir, cwd(tmpdir):
+            fake_local_fs_storage = FileSystemDataStorage()
+            fake_local_path_factory = WickerPathFactory(root_path=tmpdir)
+            with ColumnBytesFileWriter(
+                storage=fake_local_fs_storage,
+                path_factory=fake_local_path_factory,
+                target_file_rowgroup_size=10,
+            ) as writer:
+                locs = [
+                    writer.add("np_arr", FAKE_NUMPY_CODEC.validate_and_encode_object(data["np_arr"]))  # type: ignore
+                    for data in FAKE_DATA
+                ]
+
+            arrow_metadata_table = pa.Table.from_pydict(
+                {"foo": [data["foo"] for data in FAKE_DATA], "np_arr": [loc.to_bytes() for loc in locs]}
+            )
+            metadata_table_path = os.path.join(
+                tmpdir, fake_local_path_factory._get_dataset_partition_path(FAKE_DATASET_PARTITION)
+            )
+            os.makedirs(os.path.dirname(metadata_table_path), exist_ok=True)
+            papq.write_table(arrow_metadata_table, metadata_table_path)
+
+            fake_local_fs_storage.put_object(
+                serialization.dumps(FAKE_SCHEMA).encode("utf-8"),
+                fake_local_path_factory._get_dataset_schema_path(FAKE_DATASET_ID),
+            )
+            yield fake_local_fs_storage, fake_local_path_factory, tmpdir
+
+    def test_dataset_no_cache(self):
+        with self._setup_storage() as (fake_local_storage, fake_local_path_factory, tmpdir):
+            with tempfile.TemporaryDirectory() as tmp_cache_dir:
+                ds = FileSystemDataset(
+                    FAKE_NAME,
+                    FAKE_PARTITION,
+                    FAKE_VERSION,
+                    local_cache_path_prefix=None,
+                    columns_to_load=None,
+                    storage=fake_local_storage,
+                    path_factory=fake_local_path_factory,
+                    pa_filesystem=pafs.LocalFileSystem(),
+                )
+                for i in range(len(FAKE_DATA)):
+                    retrieved = ds[i]
+                    reference = FAKE_DATA[i]
+                    self.assertEqual(retrieved["foo"], reference["foo"])
+                    np.testing.assert_array_equal(retrieved["np_arr"], reference["np_arr"])
+
+                # test that nothing got copied to the non passed cache dir
+                tmpdir_size = get_size(tmp_cache_dir)
+                # assert that nothing is copied to folder that would have been used as cache
+                assert tmpdir_size == 0
+
+    def test_dataset_cached(self):
+        with self._setup_storage() as (fake_local_storage, fake_local_path_factory, tmpdir):
+            with tempfile.TemporaryDirectory() as tmp_cache_dir:
```

Comment:
Same comment as above about `tmp_cache_dir`.

Reply:
See the `local_cache_path_prefix` param below and then `tmpdir` size call at the end.